Exclude unnecessary files from packages

paescuj opened the pull request:
## Description

I noticed that there are quite a few files in the published packages which aren't necessary.
For example test files, but also whole `src` folders or files like `.editorconfig`.
This pull request updates the [files](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files) field in the `package.json` files to exclude those.

Note: Files like `README.md` and `LICENSE` are always included, therefore they don't have to be specified explicitly.

Also there are some packages with source map files. I wonder whether we really want these files in the published package or if it is sufficient if they are available in the local development environment?

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

paescuj committed:
Exclude unnecessary files from packages

paescuj committed:
Add back 'copyfiles' in api to copy yaml and liquid files to dist

rijkvanzanten committed:
Merge branch 'main' into exclude-unnecessary-files-from-packages

rijkvanzanten commented on packages/specs/package.json:
```diff
@@ -36,8 +36,6 @@
 	"files": [
 		"dist",
-		"LICENSE",
-		"README.md",
```
Isn't the readme required for rendering on npm, or does it send the readme no matter what?

paescuj replied:
Yes, the readme ([as well as other files](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files)) is always included 👍

rijkvanzanten committed:
Merge branch 'main' into exclude-unnecessary-files-from-packages

paescuj commented:
@rijkvanzanten I've noticed that there are also some packages with source map files included. Do you think we need those files in the published packages or would it be sufficient if they are available in the local development environment only?

rijkvanzanten commented:
@paescuj Lets get rid of them 👍🏻

paescuj committed:
Exclude declaration maps and all declaration files for drive packages

Excluding all declaration map files from published packages

Since the "drive" packages are only expected to be used internally, it's
enough to have the declaration files present when developing locally - no
need to include the types in the published packages

paescuj commented:
> @paescuj Lets get rid of them 👍🏻

Yup, makes sense IMO 👍

paescuj committed:
Remove unnecessary entries in "files" field of app/package.json
